fdmi/source_dock_fom.c

```diff
@@ -607,22 +607,42 @@ static int sd_fom_process_matched_filters(struct m0_fdmi_src_dock *sd_ctx,
 		fop = fop_create(src_rec, endpoint);
 		if (fop == NULL)
 			continue;
-		M0_LOG(M0_DEBUG, "will send fdmi rec");
+		M0_LOG(M0_DEBUG, "will send fop=%p fdmi rec:%p", fop, src_rec);
 		fop->f_opaque = src_rec;
 
 		/* @todo check rc and handle errors properly. */
 		rc = payload_encode(src_rec, fop);
+		M0_ASSERT(rc == 0);
 
+		/* Adding a ref. It will be dropped when reply is received. */
+		m0_fdmi__fs_get(src_rec);
```

Comment:
Actually, in original design, they are for different purposes.
But after my re-organization, they seem to be paired. So I had the same thought to do so.
But still, let me explain.

The `m0_fdmi__fs_get()` is to increase a ref on the underlying TX (transaction).
the `m0_ref_get() ` is to increase an in-memory ref count on src_rec itself.

I am thinking the `m0_ref_get() ` can be removed totally.
But, this can be a TODO in next PI, along with the failure handling.

Reply:
> I am thinking the m0_ref_get() can be removed totally.

Add TODO to not forget.